commands/backuplxc.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package commands
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/utils"
+)
+
+var backupLXCDoc = ` 
+The purpose of the backup-lxc command is to create a backup of all the
+LXC containers in a 1.25 environment.
+`
+
+func newBackupLXCCommand() cmd.Command {
+	command := &backupLXCCommand{}
+	command.remoteCommand = "backup-lxc-impl"
+	return wrap(command)
+}
+
+type backupLXCCommand struct {
+	baseClientCommand
+	backupDir string
+}
+
+func (c *backupLXCCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "backup-lxc",
+		Args:    "<environment name> <backup dir>",
+		Purpose: "create a backup of all the LXC containers for the specified environment",
+		Doc:     backupLXCDoc,
+	}
+}
+
+func (c *backupLXCCommand) Init(args []string) error {
+	args, err := c.baseClientCommand.init(args)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if len(args) == 0 {
+		return errors.New("no backup directory specified")
+	}
+	c.backupDir, args = args[0], args[1:]
+	return cmd.CheckEmpty(args)
+}
+
+func (c *backupLXCCommand) Run(ctx *cmd.Context) error {
+	if _, err := os.Stat(c.backupDir); err != nil {
+		return errors.Annotate(err, "checking backup dir")
+	}
+
+	if err := c.prepareRemote(ctx); err != nil {
+		return errors.Trace(err)
+	}
+
+	// Get a listing of all of the LXC containers in the environment.
+	var buf bytes.Buffer
+	rc, err := runViaSSH(
+		c.address,
+		c.getRemoteCommand(c.remoteCommand),
+		withStdout(&buf),
+	)
+	if err != nil {
+		return errors.Annotatef(err, "running %s via SSH", c.remoteCommand)
+	}
+	if rc != 0 {
+		return &cmd.RcPassthroughError{rc}
+	}
+	var lxcContainers lxcContainerList
+	if err := json.Unmarshal(buf.Bytes(), &lxcContainers); err != nil {
+		return errors.Trace(err)
+	}
+
+	doBackup := func(containerName, outpath string) error {
+		ctx.Infof("Backing up container %q to %s", containerName, outpath)
+		temp := outpath + ".tmp"
+		f, err := os.Create(temp)
+		if err != nil {
+			return errors.Annotate(err, "creating output file")
+		}
+		rc, err := runViaSSH(
+			c.address,
+			c.getRemoteCommand(c.remoteCommand, containerName),
+			withStdout(f),
+		)
+		f.Close()
+		if err != nil {
+			return errors.Annotatef(err, "running %s via SSH", c.remoteCommand)
+		}
+		if rc != 0 {
+			return &cmd.RcPassthroughError{rc}
+		}
+		return utils.ReplaceFile(temp, outpath)
+	}
+
+	// Create a backup of each container.
+	group, _ := errgroup.WithContext(context.Background())
```

Comment:
Actually, having stolen it there's no need for the `WithContext` call here, since you're throwing away the context (and the subtasks aren't cancellable anyway).